src/hotspot/share/oops/method.hpp

```diff
@@ -602,31 +589,30 @@ class Method : public Metadata {
   // true if method can omit stack trace in throw in compiled code.
   bool can_omit_stack_trace();
 
+  // Flags getting and setting.
+#define M_STATUS_GET_SET(name, ignore)          \
+  bool name() const { return _flags.name(); }   \
+  void set_##name(bool x) { _flags.set_##name(x); } \
+  void set_##name() { _flags.set_##name(true); }
+  M_STATUS_DO(M_STATUS_GET_SET)
+#undef M_STATUS_GET_SET
+
   // returns true if the method has any backward branches.
   bool has_loops() {
-    return access_flags().loops_flag_init() ? access_flags().has_loops() : compute_has_loops_flag();
+    return has_loops_flag_init() ? has_loops_flag() : compute_has_loops_flag();
   };
 
   bool compute_has_loops_flag();
-
-  bool has_jsrs() {
-    return access_flags().has_jsrs();
-  };
-  void set_has_jsrs() {
-    _access_flags.set_has_jsrs();
-  }
+  bool set_has_loops() { set_has_loops_flag(); set_has_loops_flag_init(); return true; }
 
   // returns true if the method has any monitors.
-  bool has_monitors() const                      { return is_synchronized() || access_flags().has_monitor_bytecodes(); }
-  bool has_monitor_bytecodes() const             { return access_flags().has_monitor_bytecodes(); }
-
-  void set_has_monitor_bytecodes()               { _access_flags.set_has_monitor_bytecodes(); }
+  bool has_monitors() const                      { return is_synchronized() || has_monitor_bytecodes(); }
 
   // monitor matching. This returns a conservative estimate of whether the monitorenter/monitorexit bytecodes
-  // propererly nest in the method. It might return false, even though they actually nest properly, since the info.
+  // properly nest in the method. It might return false, even though they actually nest properly, since the info.
   // has not been computed yet.
-  bool guaranteed_monitor_matching() const       { return access_flags().is_monitor_matching(); }
-  void set_guaranteed_monitor_matching()         { _access_flags.set_monitor_matching(); }
+  bool guaranteed_monitor_matching() const       { return monitor_matching(); }
+  void set_guaranteed_monitor_matching()         { set_monitor_matching(); }
```

Comment:
Is this method just obsolete now? If so it might be worth replacing the callers with `set_monitor_matching()` unless `set_monitor_matching()` is still meant to be private.

Reply:
The reason I left that was to anchor the comment.  There is nowhere good to put that in the X macro.  Also, didn't want to fix the callers.  It's a good point about making monitor_matching() private, but also not really doable with the X macro.  So that's why I left it.